include/openthread/diag.h

```diff
@@ -54,6 +54,9 @@ extern "C" {
 /**
  * This function processes a factory diagnostics command line.
  *
+ * The output of this function (the content written to @p aOutput) MUST terminate with `\0` and the `\0` is within the
```

Comment:
Yes. Note the implementation of `Diag::ProcessLine` and `Diag::ProcessCmd`, when the error is not `kErrorNone`, it will write an error message into the output buffer.